packages/core/src/protocol-helpers.ts

```diff
@@ -57,14 +63,10 @@ export default class ProtocolHelpers {
         tdTemplate: WoT.ExposedThingInit,
         propertyName: string
     ): void {
-        if (
-            form &&
-            tdTemplate &&
-            tdTemplate.properties &&
-            tdTemplate.properties[propertyName] &&
-            tdTemplate.properties[propertyName].forms
-        ) {
-            for (const formTemplate of tdTemplate.properties[propertyName].forms) {
+        if (form && tdTemplate.properties?.[propertyName]) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- we know that tdTemplate.properties[propertyName] is not null
+            const property = tdTemplate.properties[propertyName]!;
+            for (const formTemplate of property.forms ?? []) {
```

Comment:
I think we could also get rid of the eslint disable comment above by using something like this:

```suggestion
            for (const formTemplate of property?.forms ?? []) {
```